### ISSUE TITLE
PMPI_Bcast not found in Fortran Module if explicitly requested

### ISSUE THREAD

**Issue by SpinTensor** [open-mpi/ompi#10674]
Hi all,
I just compiled the openmpi-5.0.0rc7 version with the gcc-11.2 using the following configure line:
```bash
bash$ ../openmpi-5.0.0rc7_src/configure --prefix=/home/fuhl/software/openmpi-5.0.0rc7_gcc-11.2.0 --enable-mpi-fortran --with-tests-examples CC=gcc FC=gfortran
```
```bash
bash$ which mpif90
/home/fuhl/software/openmpi-5.0.0rc7_gcc-11.2.0//bin/mpif90
```
I encountered a problem with a small fortran test code:
```Fortran90
PROGRAM bcast
   USE, INTRINSIC :: ISO_FORTRAN_ENV
   USE mpi, ONLY : MPI_Init, MPI_INTEGER, MPI_COMM_WORLD, PMPI_Bcast, MPI_Finalize
   IMPLICIT NONE
   INTEGER, PARAMETER :: nints = 1024
   INTEGER, DIMENSION(nints) :: buffer
   INTEGER :: ierr

   CALL MPI_Init(ierr)
   buffer(:) = 1 
   CALL PMPI_Bcast(buffer, nints, MPI_INTEGER, 0, MPI_COMM_WORLD, ierr)
   CALL MPI_Finalize(ierr)
END PROGRAM bcast
```
When compiling I get the following error message:
```bash
bash$ mpif90 -o test.x test.F90
test.F90:3:57:

    3 |    USE mpi, ONLY : MPI_Init, MPI_INTEGER, MPI_COMM_WORLD, PMPI_Bcast, MPI_Finalize
      |                                                         1
Error: Symbol ‘pmpi_bcast’ referenced at (1) not found in module ‘mpi’
```
Note that I request the PMPI symbol of bcast, not the MPI, which works.
The error also disapears it the `USE mpi, ONLY : ...` is replaced by a simple `USE mpi`, thus loading all symbols.
As far as I can tell other PMPI symbols are present. Only the Bcast is missing.

**Comment by ggouaillardet** [comment]
Thanks for the report!

There are indeed two typos affecting `PMPI_Bcast()` and `PMPI_Bcast_init()` and they have been fixed in #10677.

**Comment by SpinTensor** [comment]
Thanks for the quick fix. I think we can close this issue.

**Comment by ggouaillardet** [comment]
Thanks for confirming the fix worked for you!

I am re-opening this issue and we will close it once it is merged in all the relevant release branches.

**Comment by SpinTensor** [comment]
Sorry, I guess I was a little bit quick there.

**Comment by jsquyres** [comment]
Looks like this is not an issue on v4.0.x or v4.1.x -- just v5.0.x. PR #10580 filed to cherry pick the fix to the v5.0.x branch.

**Comment by jsquyres** [comment]
Merged on main and v5.0.x -- closing this issue.  Thanks for bringing it to our attention!
